client/state/08types.go

```diff
@@ -40,7 +44,23 @@ type taskRunnerState08 struct {
 	//CreatedResources   *driver.CreatedResources
 }
 
-func (t *taskRunnerState08) Upgrade() *state.LocalState {
+type taskRunnerHandle08 struct {
```

Comment:
Can you add a `// COMPAT(0.10): Allows upgrading from 0.8.X to 0.9.0`. This will let us grep to remove upgrade path code later